modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandler.java

```diff
@@ -0,0 +1,467 @@
+   */
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, final JobContext context)
+          throws WorkflowOperationException {
+
+    final MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+    final WorkflowOperationInstance currentOperation = workflowInstance.getCurrentOperation();
+
+    final String configuredSourceFlavors = StringUtils.trimToEmpty(currentOperation.getConfiguration(SOURCE_FLAVORS_PROPERTY));
+    final String configuredSourceTags = StringUtils.trimToEmpty(currentOperation.getConfiguration(SOURCE_TAGS_PROPERTY));
+    final String configuredTargetTags = StringUtils.trimToEmpty(currentOperation.getConfiguration(TARGET_TAGS_PROPERTY));
+    final int numberOfEvents = Integer.parseInt(currentOperation.getConfiguration(NUMBER_PROPERTY));
+    final int maxNumberOfEvents = Integer.parseInt(currentOperation.getConfiguration(MAX_NUMBER_PROPERTY));
+    final String configuredPropertyNamespaces = StringUtils.trimToEmpty(currentOperation.getConfiguration(PROPERTY_NAMESPACES_PROPERTY));
+
+    if (numberOfEvents > maxNumberOfEvents) {
+      throw new WorkflowOperationException("Number of events to create exceeds the maximum of " + maxNumberOfEvents + ". Aborting.");
+    }
+
+    logger.info("Creating {} new media packages from media package with id {}.", numberOfEvents, mediaPackage.getIdentifier());
+
+    final String[] sourceTags = StringUtils.split(configuredSourceTags, ",");
+    final String[] targetTags = StringUtils.split(configuredTargetTags, ",");
+    final String[] sourceFlavors = StringUtils.split(configuredSourceFlavors, ",");
+    final String[] propertyNamespaces = StringUtils.split(configuredPropertyNamespaces, ",");
+    final String copyNumberPrefix = StringUtils.trimToEmpty(currentOperation.getConfiguration(COPY_NUMBER_PREFIX_PROPERTY));
+
+    final SimpleElementSelector elementSelector = new SimpleElementSelector();
+    for (String flavor : sourceFlavors) {
+      elementSelector.addFlavor(MediaPackageElementFlavor.parseFlavor(flavor));
+    }
+
+    final List<String> removeTags = new ArrayList<String>();
+    final List<String> addTags = new ArrayList<String>();
+    final List<String> overrideTags = new ArrayList<String>();
+
+    for (String tag : targetTags) {
+      if (tag.startsWith(MINUS)) {
+        removeTags.add(tag);
+      } else if (tag.startsWith(PLUS)) {
+        addTags.add(tag);
+      } else {
+        overrideTags.add(tag);
+      }
+    }
+
+    for (String tag : sourceTags) {
+      elementSelector.addTag(tag);
+    }
+
+    // Filter elements to copy based on input tags and input flavors
+    final Collection<MediaPackageElement> elements = elementSelector.select(mediaPackage, true);
+    final Collection<Publication> internalPublications = new HashSet<>();
+
+    for (MediaPackageElement e: mediaPackage.getElements()) {
+      if (e instanceof Publication && CHANNEL_ID.equals(((Publication) e).getChannel())) {
```

Comment:
The motivation to copy the internal publication is that we expect this operation to be used the following way:

1. Record a series of n short talks
2. Create n-1 duplicates
3. For each duplicate, open video editor to edit metadata and cut out one talk

If we would not "copy" the internal publication, a user would need to re-render the video editor previews for each of the videos which is very undesirable as this takes quite a while.

Not sure whether it is implemented that way, but the WOH should do a deep copy of the internal publication (incl. distribution of the preview tracks for the individual media packages) to avoid hidden dependencies of duplicates. We don't expect the file duplication to be a problem as this kind of recording is not expected to happen often.

@krinnewitz It does do that deep-copy thing, doesn't it?

Reply:
@staubesv yes, it should copy all the related files.